test/testclass.cpp

```diff
@@ -6488,6 +6488,31 @@ class TestClass : public TestFixture {
                    "    void f() { h(j, &s.g()); }\n"
                    "};\n");
         ASSERT_EQUALS("", errout.str());
+
+        checkConst("void j(int** x);\n"
+                   "void k(int* const* y);\n"
+                   "struct S {\n"
+                   "    int* p;\n"
+                   "    int** q;\n"
+                   "    int* const* r;\n"
+                   "    void f1() { j(&p); }\n"
+                   "    void f2() { j(q); }\n"
```

Comment:
f2 should not be const imho. it's likely the function j changes the data that q points at.

Reply:
Well, we don't warn at the moment. But the compiler would allow f2 to be const.

Reply:
Imho it's good that we don't warn ... but you have a TODO assertion that you want to warn and I am skeptic.

Reply:
I have changed the TODOs to regular asserts. But I guess this discussion will come up again at some point.